### ISSUE TITLE
Loss is nan

### ISSUE THREAD

**Issue by reeered** [PaddlePaddle/PaddleOCR#11751]
When I used the CPPD algorithm, I encountered the following errors. What could be the reason?

[2024/03/16 10:57:09] ppocr INFO: epoch: [1/100], global_step: 430, lr: 0.000195, acc: 0.000000, norm_edit_dis: 0.030599, loss: 6.949692, loss_node: 0.478943, loss_edge: 6.478870, avg_reader_cost: 0.00596 s, avg_batch_cost: 0.62916 s, avg_samples: 256.0, ips: 406.89434 samples/s, eta: 7:29:03, max_mem_reserved: 12519 MB, max_mem_allocated: 10724 MB
[2024/03/16 10:57:11] ppocr INFO: save model in ./output/exp/svtr_cppd_stn_ampO2/latest
[2024/03/16 10:57:18] ppocr INFO: epoch: [2/100], global_step: 440, lr: 0.000200, acc: 0.000000, norm_edit_dis: 0.032469, loss: nanxxx, loss_node: nanxxx, loss_edge: nanxxx, avg_reader_cost: 0.34538 s, avg_batch_cost: 0.96686 s, avg_samples: 256.0, ips: 264.77573 samples/s, eta: 7:34:20, max_mem_reserved: 12519 MB, max_mem_allocated: 10724 MB
[2024/03/16 10:57:24] ppocr INFO: epoch: [2/100], global_step: 450, lr: 0.000204, acc: 0.000000, norm_edit_dis: 0.033483, loss: nanxxx, loss_node: nanxxx, loss_edge: nanxxx, avg_reader_cost: 0.00063 s, avg_batch_cost: 0.59254 s, avg_samples: 256.0, ips: 432.03909 samples/s, eta: 7:33:28, max_mem_reserved: 12519 MB, max_mem_allocated: 10724 MB
[2024/03/16 10:57:30] ppocr INFO: epoch: [2/100], global_step: 460, lr: 0.000209, acc: 0.000000, norm_edit_dis: 0.033572, loss: 6.972978, loss_node: 0.478880, loss_edge: 6.500391, avg_reader_cost: 0.00059 s, avg_batch_cost: 0.58652 s, avg_samples: 256.0, ips: 436.46998 samples/s, eta: 7:32:32, max_mem_reserved: 12519 MB, max_mem_allocated: 10724 MB
[2024/03/16 10:57:36] ppocr INFO: epoch: [2/100], global_step: 470, lr: 0.000214, acc: 0.000000, norm_edit_dis: 0.035045, loss: nanxxx, loss_node: nanxxx, loss_edge: nanxxx, avg_reader_cost: 0.00066 s, avg_batch_cost: 0.58212 s, avg_samples: 256.0, ips: 439.77072 samples/s, eta: 7:31:35, max_mem_reserved: 12519 MB, max_mem_allocated: 10724 MB

My config:
Global:
  use_gpu: True
  epoch_num: 100
  log_smooth_window: 20
  print_batch_step: 10
  save_model_dir: ./output/exp/svtr_cppd_stn_ampO2/
  save_epoch_step: 10
  # evaluation is run every 2000 iterations after the 0th iteration
  eval_batch_step: [0, 2000]
  cal_metric_during_train: True
  pretrained_model: 
  checkpoints:
  save_inference_dir:
  use_visualdl: True
  infer_img: doc/imgs_words_en/word_10.png
  # for data or label process
  character_dict_path: ppocr/utils/ppocr_keys_v1.txt
  max_text_length: 25
  infer_mode: False
  use_space_char: False
  save_res_path: ./output/exp/predicts_svtr_cppd_stn_ampO2.txt
  use_amp: True
  amp_level: O2

Optimizer:
  name: AdamW
  beta1: 0.9
  beta2: 0.99
  epsilon: 1.e-8
  weight_decay: 0.05
  no_weight_decay_name: norm pos_embed char_node_embed pos_node_embed char_pos_embed vis_pos_embed
  one_dim_param_no_weight_decay: True
  lr:
    name: Cosine
    learning_rate: 0.001
    warmup_epoch: 5

Architecture:
  model_type: rec
  algorithm: CPPD
  Transform:
    name: STN_ON
    tps_inputsize: [32, 64]
    tps_outputsize: [32, 256]
    num_control_points: 20
    tps_margins: [0.05,0.05]
    stn_activation: none
  Backbone:
    name: PPHGNet_small

    # name: SVTRNet
    # img_size: [32, 256]
    # patch_merging: 'Conv'
    # embed_dim: [128, 256, 384]
    # depth: [6, 6, 4]
    # num_heads: [4, 8, 12]
    # mixer: ['Conv','Conv','Conv','Conv','Conv','Conv', 'Conv','Conv', 'Global','Global','Global','Global','Global','Global','Global','Global','Global','Global']
    # local_mixer: [[5, 5], [5, 5], [5, 5]]
    # last_stage: False
    # prenorm: True
  Head:
    name: CPPDHead
    dim: 384
    vis_seq: 128
    ch: &ch True

Loss:
  name: CPPDLoss
  ignore_index: &ignore_index 7000 # must be greater than the number of character classes
  smoothing: True
  sideloss_weight: 1.0

PostProcess:
  name: CPPDLabelDecode

Metric:
  name: RecMetric
  main_indicator: acc

Train:
  dataset:
    name: LMDBDataSet
    data_dir: /home/aistudio/lmdb_data/training/
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - CPPDLabelEncode: # Class handling label
          ignore_index: *ignore_index
          ch: *ch
      - SVTRRecResizeImg:
          image_shape: [3, 64, 256]
          padding: False
      - KeepKeys:
          keep_keys: ['image', 'label', 'label_node', 'label_index', 'length'] # dataloader will return list in this order
  loader:
    shuffle: True
    batch_size_per_card: &bs 256
    drop_last: True
    num_workers: 8

Eval:
  dataset:
    name: LMDBDataSet
    data_dir: /home/aistudio/lmdb_data/evaluation/
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - CPPDLabelEncode: # Class handling label
          ignore_index: *ignore_index
          ch: *ch
      - SVTRRecResizeImg:
          image_shape: [3, 64, 256]
          padding: False
      - KeepKeys:
          keep_keys: ['image', 'label', 'label_node','label_index','length'] # dataloader will return list in this order
  loader:
    shuffle: False
    drop_last: False
    batch_size_per_card: *bs
    num_workers: 2

I add conv and adaptive_pool to make HGNet Backbone output match CPPD Head input 
self.conv = nn.Conv2D(1024, 128, kernel_size=1) 
self.adaptive_pool = nn.AdaptiveAvgPool2D(output_size=[384, 1]) 

@Topdu 

BTW, could anyone tell me which algorithm (backbone, head or architecture algorithm) is better in Chinese Handwriting datasets?

**Comment by reeered** [comment]
I tried fine-tuning with the default CPPD configuration (SVTRNet), but the accuracy is always below 0.3. So I changed the backbone.

**Comment by changdazhou** [comment]
maybe you can try to set amp_level OFF

**Comment by reeered** [comment]
> maybe you can try to set amp_level OFF

How could I use amp and CPPD at the same time?

**Comment by Topdu** [comment]
Please try the following settings:
 use_amp: True
 amp_level: O2
 amp_custom_black_list: ['exp', 'softmax']

**Comment by reeered** [comment]
> Please try the following settings: use_amp: True amp_level: O2 amp_custom_black_list: ['exp', 'softmax']

Thanks for your reply! I will have a try.
